### DISCUSSION TITLE
Import directus utils/classes in extensions

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#10154] [Archive]
Hi all,

I'm having a linting issue which I can't seem to resolve quickly. I'm trying to import a Directus class (Url) in my notification extension.

The import looks like this:
```
const { Url } = require('directus/utils/url');
```

But this throws a no-resolved eslinting error. What's the best way to import such entities? Or should I disable this no-resolve error 😏 

Cheers!

**Comment by nickrum** [comment]
Looks like the ESLint import plugin doesn't yet support the `exports` package.json field (see [here](https://github.com/import-js/eslint-plugin-import/issues/1868)). I don't think there is anything we can do except for waiting for the plugin to add support for it. In the meantime you can indeed disable the rule or suppress the error for this specific line.

**Reply by keesvanbemmel** [reply]
Thanks @nickrum! @rijkvanzanten beat me to the marking part ;)

**Comment by NeoKactus** [comment]
Thank you so much. I was having a lot of trouble getting the import just right.

The utils that use `export default` also caused me a [headache](https://stackoverflow.com/questions/33704714/cant-require-default-export-value-in-babel-6-x). For anyone else coming across this discussion, this is how you can import them: 
`const isUrlAllowed = require('directus/utils/is-url-allowed').default;`


PHPStorm still resolves with `import isUrlAllowed from "directus/dist/utils/is-url-allowed";`, but I can work around that. :)

**Reply by rijkvanzanten** [reply]
@King-Aardvark the whole Node CJS / ESM in combination with TypeScript situation is a massive headache.. I've actually just updated the whole app codebase to effectively never rely on default exports for this exact reason. We might try dual exporting from the api utils so you can do with or without `.default`, but that needs some more research

**Reply by NeoKactus** [reply]
That sounds great. It would be definitely make the imports more consistent. 

Glad to hear it's on your radar. I appreciate the effort the Directus team is putting in.
